tests/functional/DatadirTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\GoodDataWriter\Tests\Functional;
+
+use Keboola\DatadirTests\AbstractDatadirTestCase;
+use Keboola\DatadirTests\DatadirTestSpecification;
+use Keboola\GoodData\Client;
+use Keboola\GoodDataWriter\Test\ApiHelper;
+use Keboola\StorageApi\Components;
+use Keboola\StorageApi\Options\Components\Configuration;
+
+class DatadirTest extends AbstractDatadirTestCase
+{
+    /** @var array */
+    protected $config;
+    /** @var Client */
+    protected $gdClient;
+
+    public function setup(): void
+    {
+        parent::setUp();
+        $fp = file_get_contents(__DIR__.'/../config.json');
+        if ($fp === false) {
+            throw new \Exception('config.json not found');
+        }
+        $this->config = \GuzzleHttp\json_decode($fp, true);
+        $this->config['parameters']['user'] = [
+            'login' => getenv('GD_USERNAME'),
+            '#password' => getenv('GD_PASSWORD'),
+        ];
+        $this->config['parameters']['project'] = [
+            'pid' => getenv('GD_PID'),
+        ];
+
+        $this->gdClient = new Client('https://secure.gooddata.com');
+        $this->gdClient->login(getenv('GD_USERNAME'), getenv('GD_PASSWORD'));
+    }
+
+
+    public function testRun(): void
+    {
+        ApiHelper::cleanUpProject($this->gdClient, (string) getenv('GD_PID'));
+
+        // Run
+        $config = $this->config;
+        $config['action'] = 'run';
+
+        $specification = new DatadirTestSpecification(
+            __DIR__ . '/run/source/data',
+            0,
+            '',
+            '',
+            __DIR__ . '/run/expected/data/out'
+        );
+        $tempDatadir = $this->getTempDatadir($specification);
+        file_put_contents($tempDatadir->getTmpFolder() . '/config.json', \GuzzleHttp\json_encode($config));
+        $process = $this->runScript($tempDatadir->getTmpFolder());
+        $this->assertMatchesSpecification($specification, $process, $tempDatadir->getTmpFolder());
+
+        $res = $this->gdClient->get('/gdc/md/' . getenv('GD_PID') . '/data/sets');
+        $this->assertCount(5, $res['dataSetsInfo']['sets']);
+        $this->assertArrayHasKey('lastSuccess', $res['dataSetsInfo']['sets'][0]);
+        // Assert that last data load occured within a minute
+        $this->assertTrue(time() < 60 + strtotime($res['dataSetsInfo']['sets'][0]['lastSuccess']));
+
+
+        // Read model
```

Comment:
nemohly by to byt dva testy?

Reply:
Mohly, ale takhle je to jednodušší, protože první test vyrobí v GD strukturu pro ten druhý test. Když jde jen o základní func testy, tak by ta závislost snad nemusela vadit? ;o)

Reply:
no jo, uz to vidim, a tak ze ses to ty, tak by mohlo :)

co ty datadir testy, byl to pain?

Reply:
Kua, až na to, že lokálně mi prochází a v Travisu vyfailovaly už asi popátý, tak to bylo docela v pohodě. 😃

Reply:
ti dotedka failovaly normalni testy :)

Reply:
No ony si tam nějak vzájemně rozházely GD projekt, teď si zase lezou do zelí s temp souborama i když by každý měl mít svůj, tak snad nějak postupně dopivotuju k zelené. ;o)

Reply:
no ten temp si budes muset vymazat rucne, kdyz mas dva testy splacly v jedne metode